src/Microsoft.Sbom.Api/Config/ConfigSanitizer.cs

```diff
@@ -45,6 +45,13 @@ public IConfiguration SanitizeConfig(IConfiguration configuration)
                 .WriteTo.Console(outputTemplate: Constants.LoggerTemplate)
                 .CreateLogger();
 
+            // If BuildDropPath is null then run the logic to check whether it is required or not based on the current configuration.
+            if (configuration.BuildDropPath?.Value == null || (configuration.DockerImagesToScan?.Value != null && configuration.BuildComponentPath?.Value == null))
+            {
+                ValidateBuildDropPathConfiguration(configuration);
+                configuration.BuildDropPath = GetTempBuildDropPath(configuration);
```

Comment:
Why is this temporary path needed? Can we just ignore providers that use `BuildDropPath` if  `BuildDropPath` is not provided? We should still enforce valid arguments from the client by arg checking, but on workflow/provider level, we can potentially just ignore and continue to the next provider if `BuildDropPath` is empty. Though we can add logging that says something like _"No `BuildDropPath` provided, skipping"_ in these providers.